textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/fuzzytokenmatcher/FuzzyTokenMatcher.java

```diff
@@ -28,16 +36,61 @@ public FuzzyTokenMatcher(IPredicate predicate) {
     
     @Override
     public void open() throws DataFlowException {
-    	return;
+    	try {
+            sourceOperator.open();
+            attributeList = predicate.getAttributeList();
+            threshold = predicate.getThreshold();
+            queryTokens = predicate.getQueryTokens();
+    	} catch (Exception e) {
+            e.printStackTrace();
+            throw new DataFlowException(e.getMessage(), e);
+    	}
     }
 
     @Override
     public ITuple getNextTuple() throws DataFlowException {
-	return null;
+		try {
+		    ITuple sourceTuple = sourceOperator.getNextTuple();
+		    if (sourceTuple == null || !this.predicate.getisSpanInformationAdded())
+		        return sourceTuple;
+            
+		    int schemaIndex = sourceTuple.getSchema().getIndex(SchemaConstants.SPAN_LIST_ATTRIBUTE.getFieldName());
+		    List<Span> resultSpans =
+                    (List<Span>)sourceTuple.getField(schemaIndex).getValue();
+            
+		    for(int attributeIndex = 0; attributeIndex < attributeList.size(); attributeIndex++) {
+		        String fieldName = attributeList.get(attributeIndex).getFieldName();
+		        IField field = sourceTuple.getField(fieldName);
+                
+		        if (field instanceof TextField) {
```

Comment:
What if it's a STRING type?  Can we use the analyzer to tokenize it then do the match?